packages/template-retail-react-app/app/components/swatch-group/swatch.jsx

```diff
@@ -18,32 +18,30 @@ import {Link as RouteLink} from 'react-router-dom'
 /**
  * The Swatch Component displays item inside `SwatchGroup`
  */
-const Swatch = (props) => {
-    const {
-        disabled,
-        selected,
-        label,
-        children,
-        href,
-        variant = 'square',
-        onChange,
-        value,
-        name
-    } = props
+const Swatch = ({
+    children,
+    disabled,
+    href,
+    label,
+    name,
+    selected,
+    onKeyDown,
+    variant = 'square'
+}) => {
     const styles = useMultiStyleConfig('SwatchGroup', {variant, disabled, selected})
     return (
         <Button
             {...styles.swatch}
             as={RouteLink}
             to={href}
             aria-label={name}
-            onClick={(e) => {
-                e.preventDefault()
-                onChange(value, href)
-            }}
             aria-checked={selected}
             variant="outline"
             role="radio"
+            onKeyDown={onKeyDown}
+            // To mimic the behavior of native radio inputs, only the selected input should be
+            // tabbable. (The rest are selectable via arrow keys.)
+            tabindex={selected ? 0 : -1}
```

Comment:
Aside: the native behavior fields awkward in terms of accessibility since the non-selected options are not tabable

Reply:
It makes sense if you think of the focused component as the radio _group_, rather than the individual radio button. Compare with a dropdown, where tab moves focus on/off the `<select>`, while arrow keys are how you choose an `<option>`.